tests/validation/tests/v3_api/resource/rbac/monitoring/monitoring_rbac.json

```diff
@@ -0,0 +1,207 @@
+      "get configmap test-example-1",
+      "apply  -f {resource_root}/rbac/monitoring/cr_configmap_upgrade.yaml",
+      "delete -f {resource_root}/rbac/monitoring/cr_configmap_upgrade.yaml",
+      "create -f {resource_root}/rbac/monitoring/cr_secret_base.yaml",
+      "get secret",
+      "get secret test-example-1",
+      "apply  -f {resource_root}/rbac/monitoring/cr_secret_upgrade.yaml",
+      "delete -f {resource_root}/rbac/monitoring/cr_secret_base.yaml",
+      "create -f {resource_root}/rbac/monitoring/cr_pod_monitor_base.yaml",
+      "get podmonitors.monitoring.coreos.com",
+      "get podmonitors.monitoring.coreos.com test-example-1",
+      "apply  -f {resource_root}/rbac/monitoring/cr_pod_monitor_upgrade.yaml",
+      "delete -f {resource_root}/rbac/monitoring/cr_pod_monitor_upgrade.yaml",
+      "create -f {resource_root}/rbac/monitoring/cr_prometheus_rule_base.yaml",
+      "get prometheusrules.monitoring.coreos.com",
+      "get prometheusrules.monitoring.coreos.com test-example-1",
+      "apply  -f {resource_root}/rbac/monitoring/cr_prometheus_rule_upgrade.yaml",
+      "delete -f {resource_root}/rbac/monitoring/cr_prometheus_rule_upgrade.yaml",
+      "create -f {resource_root}/rbac/monitoring/cr_service_monitor_base.yaml",
+      "get servicemonitors.monitoring.coreos.com",
+      "get servicemonitors.monitoring.coreos.com test-example-1",
+      "apply  -f {resource_root}/rbac/monitoring/cr_service_monitor_upgrade.yaml",
+      "delete -f {resource_root}/rbac/monitoring/cr_service_monitor_upgrade.yaml",
+      "get alertmanagers.monitoring.coreos.com",
+      "get alertmanagers.monitoring.coreos.com test-example-1",
+      "get prometheuses.monitoring.coreos.com",
+      "get prometheuses.monitoring.coreos.com test-example-1"
+    ],
+    "should_fail": [
+      "create -f {resource_root}/rbac/monitoring/cr_alert_manager.yaml",
+      "apply  -f {resource_root}/rbac/monitoring/cr_alert_manager.yaml",
+      "delete -f {resource_root}/rbac/monitoring/cr_alert_manager.yaml",
+      "create -f {resource_root}/rbac/monitoring/cr_prometheus.yaml",
+      "apply  -f {resource_root}/rbac/monitoring/cr_prometheus.yaml",
+      "delete -f {resource_root}/rbac/monitoring/cr_prometheus.yaml"
+    ]
+  },
+  "monitoring-view": {
+    "should_pass": [
+      "get configmap",
+      "get configmap my-config",
+      "get secret",
+      "get secret mysecret",
+      "get podmonitors.monitoring.coreos.com",
+      "get podmonitors.monitoring.coreos.com test-example-1",
+      "get prometheusrules.monitoring.coreos.com",
+      "get prometheusrules.monitoring.coreos.com test-example-1",
+      "get servicemonitors.monitoring.coreos.com",
+      "get servicemonitors.monitoring.coreos.com test-example-1",
+      "get alertmanagers.monitoring.coreos.com",
+      "get alertmanagers.monitoring.coreos.com test-example-1",
+      "get prometheuses.monitoring.coreos.com",
+      "get prometheuses.monitoring.coreos.com test-example-1"
```

Comment:
Each role's suite of test cases seems to rely on the previous role having successfully passed each test case in order to return the correct results.

For example:
1. `monitoring-view` tests rely on the tester having previously (and successfully) ran either `monitoring-admin` or `monitoring-edit` tests to create the resources in the first place
2. `monitoring-edit` relies on `monitoring-admin` being able to successfully delete resources so that its create doesn't fail because the resource already exists (even if they do have the permissions to create).

If there is a requirement for these roles to be independently testable (e.g. wanting to run tests for roles in parallel or individually), there should be another field to the test cases that runs a set of commands to add or remove resources from cluster before running the tests for a given role.

Not sure if this is something that needs to be addressed, feel free to resolve this comment if this requirement is not necessary.

Reply:
We are more focused on validating if the cluster role gets the right set of permissions, rather than if the object is created/updated/got/listed/deleted with target values. That is also why we check the particular `Error from server (Forbidden)` error in the response of API (kubectl command) calls.